socketio/src/client/raw_client.rs

```diff
@@ -306,51 +319,95 @@ impl RawClient {
     /// Handles a binary event.
     #[inline]
     fn handle_binary_event(&self, packet: &Packet) -> Result<()> {
-        let event = if let Some(string_data) = &packet.data {
-            string_data.replace('\"', "").into()
-        } else {
-            Event::Message
+        let event = match &packet.data {
+            Some(Value::String(e)) => Event::from(e.to_owned()),
+            Some(Value::Array(array)) => match array.first() {
+                Some(Value::String(e)) => Event::from(e.to_owned()),
+                _ => Event::Message,
+            },
+            _ => Event::Message,
         };
 
-        if let Some(attachments) = &packet.attachments {
-            if let Some(binary_payload) = attachments.get(0) {
-                self.callback(&event, Payload::Binary(binary_payload.to_owned()))?;
+        let payload = match &packet.data {
+            Some(Value::Array(vec)) if vec.len() == 1 => {
```

Comment:
nit: clippy lint https://rust-lang.github.io/rust-clippy/master/#len_zero
```suggestion
            Some(Value::Array(vec)) if ! vec.is_empty() => {
```

Reply:
here is len() == 1

Reply:
Yes hence why it's `! vec.is_empty()`. is_empty() is as faster operation than len()

Reply:
I got that, if not empty, get(0) will not panic.